Sources/SwiftProtobuf/Message+BinaryAdditions.swift

```diff
@@ -34,18 +34,15 @@ public extension Message {
     let requiredSize = try serializedDataSize()
     var data = Data(count: requiredSize)
     try data.withUnsafeMutableBytes { (pointer: UnsafeMutablePointer<UInt8>) in
-      try serializeBinary(into: pointer)
+      var visitor = BinaryEncodingVisitor(forWritingInto: pointer)
+      try traverse(visitor: &visitor)
+      if requiredSize != visitor.encoder.distance(pointer: pointer) {
```

Comment:
Agreed—don't give users the opportunity to catch our bugs. (It's also easy to silently ignore errors using `try?`, and we'd want this to show up so people let us know.)